content/docs/api-reference/dvcfilesystem.md

```diff
@@ -0,0 +1,126 @@
+# DVCFileSystem
+
+_New in DVC 2.27_
+
+DVCFileSystem provides a pythonic file interface (
+[fsspec-compatible](https://filesystem-spec.readthedocs.io/)) for a DVC repo. It
+is a read-only filesystem, hence it does not support any write operations, like
+`put_file`, `cp`, `rm`, `mv`, `mkdir` etc.
+
+DVCFileSystem provides a unified view of all the files/directories in your
+repository, be it Git-tracked or DVC-tracked, or untracked (in case of a local
+repository). It can reuse the files in DVC <abbr>cache</abbr> and can otherwise
+stream from
+[supported remote storage](/doc/command-reference/remote/add#supported-storage-types).
+
+```py
+>>> from dvc.api import DVCFileSystem
+# opening a local repository
+>>> fs = DVCFileSystem("/path/to/local/repository")
+# opening a remote repository
+>>> url = "https://github.com/iterative/example-get-started.git"
+>>> fs = DVCFileSystem(url, rev="main")
+```
+
+The optional positional argument can be a URL or a local path to the DVC
+project. If unspecified, the DVC project in current working directory is used.
+
+The optional `rev` argument can be passed to open a filesystem from a certain
+Git commit (any [revision](https://git-scm.com/docs/revisions) such as a branch
+or a tag name, a commit hash, or an [experiment name]).
+
+[experiment name]: /doc/command-reference/exp/run#-n
```

Comment:
Would it be better to put the [class signature](https://github.com/iterative/dvc/blob/391bb29be29fe9ac3036180f503590c97b71c850/dvc/fs/dvc.py#L368) up somewhere? With param names, etc. (may need to make a mixed definition combined with [FileSystem](https://github.com/iterative/dvc/blob/570e3a5359b41520ea01952df11602e40d69f793/dvc/objects/fs/base.py#L55))